--- /home/ngrinszt/anaconda3/envs/GRAN2/lib/python3.8/site-packages/torch/nn/modules/linear.py
+++ /home/ngrinszt/anaconda3/envs/GRAN2/lib/python3.8/site-packages/torch/nn/modules/linear.py
@@ -31,7 +31,7 @@
         >>> print(output.size())
         torch.Size([128, 30])
     """
-    __constants__ = ['bias', 'in_features', 'out_features']
+    __constants__ = ['in_features', 'out_features']
 
     def __init__(self, in_features, out_features, bias=True):
         super(Linear, self).__init__()