--- /home/ngrinszt/HPC/model.py
+++ /home/ngrinszt/HPC/model.py
@@ -9,7 +9,7 @@
         self.listmlp = nn.ModuleList()
         self.listmlp_pass = nn.ModuleList()
         self.listmlp_value = nn.ModuleList()
-        self.listgcn.append(BaseConvHeterogene(input_dim, hidden_dim, 'gcn', withbn=withbn))
+        self.listgcn.append(BaseConvHeterogene(input_dim, hidden_dim, 'gcn', res=res, withbn=withbn))
         for _ in range(ngcn-1):
             self.listgcn.append(BaseConvHeterogene(hidden_dim, hidden_dim, 'gcn', res=res, withbn=withbn))
         for _ in range(nmlp-1):
@@ -19,7 +19,7 @@
             self.listmlp_value.append(BaseConvHeterogene(hidden_dim, hidden_dim, 'mlp', res=res, withbn=withbn))
         self.listmlp_value.append(Linear(hidden_dim, 1))
 
-        self.listmlp_pass.append(BaseConvHeterogene(hidden_dim+3, hidden_dim, 'mlp', withbn=withbn))
+        self.listmlp_pass.append(BaseConvHeterogene(hidden_dim+3, hidden_dim, 'mlp', res=res, withbn=withbn))
         for _ in range(nmlp-2):
             self.listmlp_pass.append(BaseConvHeterogene(hidden_dim, hidden_dim, 'mlp', res=res, withbn=withbn))
         self.listmlp_pass.append(Linear(hidden_dim, 1))